tutorials/blitz/7_captum_explainability.py

```diff
@@ -0,0 +1,203 @@
+#
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
+model = GCN(g.ndata['feat'].shape[1], 16, dataset.num_classes).to(device)
+g = g.to(device)
+optimizer = torch.optim.Adam(model.parameters(), lr=0.01, weight_decay=5e-4)
+
+features = g.ndata['feat']
+labels = g.ndata['label']
+train_mask = g.ndata['train_mask']
+
+for epoch in range(1,201):
+    model.train()
+    optimizer.zero_grad()
+    # Forward
+    logits = model(features, g)
+    
+    # Compute prediction
+    pred = logits.argmax(1)
+    
+    # Compute loss
+    # Note that you should only compute the losses of the nodes in the training set.
+    loss = F.cross_entropy(logits[train_mask], labels[train_mask])
+    
+    # Backward
+    loss.backward()
+    optimizer.step()
+
+######################################################################
+# Explaining the predictions
+# --------------------------
+# First, we will complete the task of which input nodes are more important.
+# We attribute the model predictions to the input node features with IntegratedGradients.
+#
+
+# Select the node with index 10 for interpretability analysis
+output_idx = 10
+target = int(g.ndata['label'][output_idx])
+print(target)
+
+# Since the ``IntergratedGradients`` method only allows one argument to be passed, we use ``partial`` function to pass the default value to the forward function.
+#
+
+# import captum
+from captum.attr import IntegratedGradients
+from functools import partial
+
+# Node explainability
+ig = IntegratedGradients(partial(model.forward, g=g))
+# Attribute the predictions for node class 0 to the input features
+ig_attr_node = ig.attribute(g.ndata['feat'], target=target,
+                            internal_batch_size=g.num_nodes(), n_steps=50)
+print(ig_attr_node.shape)
+
+
+# We compute the node importance weights from the input feature weights and normalize them.
+#
+
+# Scale attributions to [0, 1]:
+ig_attr_node = ig_attr_node.abs().sum(dim=1)
+ig_attr_node /= ig_attr_node.max()
+
+
+# We visualize node-centered weighted subgraphs through DGL's built-in functions.
+#
+
+# Visualize
+from utility import visualize_subgraph
+import matplotlib.pyplot as plt
+
+num_hops = 2
+ax, nx_g = visualize_subgraph(g, output_idx, num_hops, node_alpha=ig_attr_node)
+plt.show()
+
+# <img src="README.assets/Figure_1.png" style="zoom:50%;" />
+# Then, we will complete the task of which input edges are more important.
+# To apply the IntergratedGradients method, we redefine the forward function
+#
+
+def model_forward(edge_mask, g):
+    out = model(g.ndata['feat'],g,edge_weight=edge_mask)
+    return out
+
+# Edge explainability
+edge_mask = torch.ones(g.num_edges()).requires_grad_(True).to(device)
+ig = IntegratedGradients(partial(model_forward, g=g))
+ig_attr_edge = ig.attribute(edge_mask, target=target,
+                            internal_batch_size=g.num_nodes(), n_steps=50)
```

Comment:
Should `internal_batch_size` still be `g.num_nodes()`? Do we need to pass a value for this argument?